project/MiMa.scala

```diff
@@ -1131,6 +1131,11 @@ object MiMa extends AutoPlugin {
         ProblemFilters.exclude[IncompatibleResultTypeProblem]("akka.cluster.sharding.ShardRegion.shardBuffers"),
         ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.cluster.sharding.ShardRegion.shardBuffers_="),
 
+        // #18470 Calls to stash() inside of persist callbacks should throw exception
```

Comment:
Ah, no, just rebase this PR onto lastest master and move these MiMa exceptions to the latest version instead of under 2.4.16 where they are now.

The question whether this PR breaks binary compatiblity is a separate one: if it breaks binary compatibility in a problematic way it'll have to move to akka 3, but it looks like it might not. Would be good to get another pair of eyes on that though.